gitpod-io/gitpod
# [public-api] Measure incoming JWT Sessions - WEB-102

## Description


Adds a new interceptor, which runs last in our chain, which checks the type of token that was accepted by the authentication middleware and reports metrics for whether it contained a JWT session cookie

## Related Issue(s)

 * Depends on https://github.com/gitpod-io/gitpod/pull/17332

## How to test

1. Preview
2. Sign in
3. Access dashboard which makes Public Api server calls
4. No failed requests
5. Port forward to the public api `metrics` port, observe `requests_with_jwt_sessions_total` present

## Documentation


#### Preview status

gitpod:summary

## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
